Controller/AbstractCrudController.php

```diff
@@ -165,6 +166,17 @@ private function getEntityValidator()
      */
     private function getDataMapper()
     {
+        $config = $this->getPreparedConfig();
```

Comment:
If improving config resolving logic then it makes sense to add it right to **getConfiguredService** method itself, then all other parts of code will benefit from it as well. This is what I have in mind:

```
/**
 * @param string $serviceType
 *
 * @return object
 */
private function getConfiguredService($serviceType)
{
    $config = $this->container->getParameter(ModeraServerCrudExtension::CONFIG_KEY);

    try {
        $serviceId = $config[$serviceType];

        return $this->container->get($serviceId);
    } catch (\Exception $e) {
        $configException = BadConfigException::create($serviceType, $config, $e);

        throw $e;
    }
}
```

It requires adding of additional exception class **BadConfigException** which can be put to Exceptions namespace of the bundle. First two parameters passed to static factory **create** can be accessed with getter methods of exception. By passing an original exception as a third argument developers will be able to pinpoint the reason why exception has occurred as well as get access to normalized data of exception using gettrs - **getServiceType**, **getServiceId**. A sample exception message could look similar to something like this:

**An error occurred while getting a service for configuration property "data_mapper" using DI service with ID "acme.my_foo_mapper" - [original exception message]**. 

[original exception message] can be replaced with a message from a given $e as a third parameter of **create** factory method.